Starfinder 2e Dark Theme Stylesheet

LiftVortex opened the pull request:
Update:
- I've updated the files and structure to be the same as the newest main commit
- Included a Dark Mode option in the settings toggle to use this theme.
- Update the en.json to include Dark.
- Changed the logo to match the new commit.

Fork includes:
- Stylesheet which customises the PC Sheet.

stwlam commented:
Would you mind reducing this down to the PC sheet so it can be considered in isolation?

LiftVortex commented:
> Would you mind reducing this down to the PC sheet so it can be considered in isolation?

Yeah no problem, I'll split it into two style sheets.

LiftVortex commented:
> Would you mind reducing this down to the PC sheet so it can be considered in isolation?

I've now updated the pull request to just the PC Sheet.

LiftVortex commented:
@stwlam I've now updated the files and structure to be the same as the newest main commit which included the light mode option.

Changes below:
- Included a Dark Mode option in the settings toggle to use this theme.
- Update the en.json to include Dark.
- Changed the logo to match the new commit.
- Removed the attribute builder and pop outs theming.
- Updated sf2e.mjs as it wasn't changing between styles.

LiftVortex commented:
I've completely rewritten the dark.css stylesheet to be inline with the style and formatting of light.css.
I've also removed asset that are no longer used and renamed the remaining ones.

stwlam commented on module.json:
```diff
@@ -37,8 +37,13 @@
             "flags": {}
         }
     ],
+    "styles": [
```
Don't include these

LiftVortex replied:
I've now removed these.

stwlam commented on module.json:
```diff
@@ -37,8 +37,13 @@
     "scripts": [
-        "sf2e.mjs"
+        "sf2e.mjs", 
+        "scripts/sf2e-Overrides.js"
```
ja dom alterations are going to be too fragile for something like this

LiftVortex replied:
I've now removed this file.

stwlam commented on sf2e.mjs:
```diff
@@ -2,7 +2,10 @@ const MODULE_ID = "starfinder-field-test-for-pf2e";
 function applyTheme(themeId) {
   const headEl = document.head;
-  headEl.querySelector(`link[href*=${MODULE_ID}]`)?.remove();
+
```
This isn't necessary

LiftVortex replied:
I've reverted this to how it was previously.

stwlam commented on sf2e.mjs:
```diff
@@ -25,10 +28,12 @@ Hooks.once("init", () => {
       },
       initial: "none",
     }),
+    default: "none",
```
This `initial` is setting the default

LiftVortex replied:
I've removed this setting.

LiftVortex commented:
@stwlam thanks for the comments.
I've updated the fork to resolve these issues.
Apologies, I only started learning CSS and JavaScript for this project, still got a lot of learning to do. Thanks to your comments I now understand why I was having issues previously!

TikaelSol commented on module.json:
```diff
@@ -751,4 +751,4 @@
         }
     }
-}
+}
```
```suggestion
}

```

TikaelSol commented on sf2e.mjs:
```diff
@@ -37,4 +38,4 @@ Hooks.once("init", () => {
 Hooks.once("setup", () => {
   applyTheme(game.settings.get(MODULE_ID, "theme"));
-});
+});
```
```suggestion
});

```

TikaelSol commented on styles/dark.css:
```diff
@@ -0,0 +1,771 @@
+      }
+    }
+  }
+}
```
```suggestion
}

```

LiftVortex commented:
@TikaelSol @stwlam , I've updated the fork to resolve the conflicts with the main branch.
I've also fixed some style bugs I've found when testing.
Sorry about the line endings, I didn't notice that GitHub Desktop automatically changes them to CRLF when you commit, I've fixed that now.

TikaelSol commented on styles/dark.css:
```diff
@@ -0,0 +1,857 @@
+      }
+    }
+  }
+}
```
```suggestion
}

```

Yeah, these are a pain🙃
